python/ray/data/dataset.py

```diff
@@ -622,6 +624,46 @@ def map_batches(
                 "``compute=ActorPoolStrategy(min, max)``."
             )
 
+        if isinstance(fn, Predictor):
+            from ray.air.checkpoint import Checkpoint
+            from ray.train.batch_predictor import BatchPredictor
+
+            class _WrapperPredictor(Predictor):
+                def __init__(self, base_predictor: Predictor):
+                    self.base_predictor = base_predictor
+
+                @classmethod
+                def from_checkpoint(cls, checkpoint):
+                    return cls(base_predictor=checkpoint.to_dict()["predictor"])
+
+                def predict(self, data, **kwargs):
+                    return self.base_predictor.predict(data, **kwargs)
+
+                def _predict_pandas(self, data, **kwargs):
+                    return self.base_predictor._predict_pandas(data, **kwargs)
+
+                def _predict_numpy(self, data, **kwargs):
+                    return self.base_predictor._predict_numpy(data, **kwargs)
+
+            checkpoint = Checkpoint.from_dict({"predictor": fn})
+
+            batch_predictor = BatchPredictor.from_checkpoint(
+                checkpoint, predictor_cls=_WrapperPredictor
+            )
+
+            assert isinstance(compute, ActorPoolStrategy)
+            kwargs = fn_kwargs if fn_kwargs else {}
+            return batch_predictor.predict(
```

Comment:
Let's get rid of this dependency on BatchPredictor? We want to deprecate that class.